docker-machine integration

mikes1988 opened the pull request:
PR to add docker-machine to the default image.  This could potentially be useful in an environment where a sysadmin wants to manage a group of docker hosts.

mikes1988 committed:
Adding docker-machine into the docker image.

Potentially useful for situations where there are a number of docker hosts needing maintained/controlled.
Also added a quick note in README.md relating to this.

mikes1988 committed:
We actually don't need curl anymore, this was actually from a previous test I had completed.

mikes1988 committed:
Adding environment variable for docker machine - make the MACHINE_STORAGE_PATH a persistent path.

gimoh commented on Dockerfile:
```diff
@@ -27,6 +27,11 @@ RUN mkdir /etc/devops-utils
 ADD . /opt/devops-utils
 RUN cd /opt/devops-utils && pip install . && \
     cp -r init_plugins runner_plugins /etc/devops-utils/
+ADD ["https://github.com/docker/machine/releases/download/v0.3.0/docker-machine_linux-amd64", \
```
Could you move this up maybe just after `pip install Fabric` line?  This will prevent the binary from getting downloaded even if only source directory changed (i.e. normally the `ADD . /opt/devops-utils` will invalidate cache for every following instruction).

mikes1988 committed:
Update Dockerfile

Moving docker-machine 'add' higher up in the file, prevent downloading the binary when there has only been a minor change in later build steps.

mikes1988 commented:
Have modified the PR, never thought about that lol.

gimoh commented on Dockerfile:
```diff
@@ -27,6 +27,11 @@ RUN mkdir /etc/devops-utils
+ADD ["https://github.com/docker/machine/releases/download/v0.3.0/docker-machine_linux-amd64", \
+     "/usr/local/bin/docker-machine"]
+RUN chmod +x /usr/local/bin/docker-machine
+ENV MACHINE_STORAGE_PATH=/opt/devops-utils/docker-machine
```
Hmm, given that docker-machine would be mostly usable in `++dev` mode (so its config changes are preserved), I'm thinking to maybe set the storage path to e.g. `/opt/app/.docker/machine`.  This way it will do The Right Thing (tm) when ran in `++dev` mode, or even when ran with `--volume=$HOME:/opt/app` (just for kicks ;-) (and compat with docker-machine defaults)).

gimoh commented:
Thanks for the PR, reviewing it now :)

gimoh commented on Dockerfile:
```diff
@@ -21,12 +21,17 @@ MAINTAINER gimoh <gimoh@bitmessage.ch>
 RUN cd /opt/devops-utils && pip install . && \
     cp -r init_plugins runner_plugins /etc/devops-utils/
+RUN chmod +x /usr/local/bin/docker-machine
+ENV MACHINE_STORAGE_PATH=/opt/devops-utils/docker-machine
```
This got attached to that old diff, so here it is again:

Hmm, given that docker-machine would be mostly usable in `++dev` mode (so its config changes are preserved), I'm thinking to maybe set the storage path to e.g. `/opt/app/.docker/machine`.  This way it will do The Right Thing (tm) when ran in `++dev` mode, or even when ran with `--volume=$HOME:/opt/app` (just for kicks ;-) (and compat with docker-machine defaults)).

Also, you can move all the instructions for d-m up, keep them together and maybe add some blank lines between them and the other bits for readability.

gimoh commented:
Yip, just tested with `docker run -it --rm --volume=$HOME:/opt/app --env=MACHINE_STORAGE_PATH=/opt/app/.docker/machine devops-utils-dm docker-machine ls` and it picks up my machines from host.

OK, so just that minor change and this is ready to go.

If you're happy enough with that could you also add an example command (using `devops-utils ++dev docker-machine ls` even) to `docs/usage.rst` (maybe a subheader for d-m under Running header or sth)?  Just so there is some record of how to use it.  Maybe also mention that you need to use `++dev` or the changes will disappear lol

mikes1988 committed:
Moving docker-machine configuration to keep it all together for readability.

mikes1988 committed:
docker-machine documentation & usage example.

gimoh commented:
Brilliant, thanks a lot :)
